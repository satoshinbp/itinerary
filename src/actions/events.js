import db from '../firebase/firebase'

// ADD_EVENT
export const addEvent = event => ({
  type: 'ADD_EVENT',
  event
})

export const startAddEvent = (tripId, eventData) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid
    const eventsRef = db.collection('users').doc(uid).collection('trips').doc(tripId).collection('events')
    const { title, date, startTime, endTime, location, note } = eventData
    const event = { title, date, startTime, endTime, location, note, user: uid, tripId }

    return eventsRef.add(event).then(snapshot => (
      dispatch(addEvent({
        id: snapshot.id,
        ...event
      }))
    ))
  }
}

// EDIT_EVENT
export const editEvent = (id, updates) => ({
  type: 'EDIT_EVENT',
  id,
  updates
})

export const startEditEvent = (tripId, id, updates) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid
    const eventsRef = db.collection('users').doc(uid).collection('trips').doc(tripId).collection('events').doc(id)
    return eventsRef.update(updates).then(() => dispatch(editEvent(id, updates)))
  }
}

// REMOVE_EVENT
export const removeEvent = id => ({
  type: 'REMOVE_EVENT',
  id
})

export const startRemoveEvent = (tripId, id) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid
    const eventsRef = db.collection('users').doc(uid).collection('trips').doc(tripId).collection('events').doc(id)
    return eventsRef.delete().then(() => dispatch(removeEvent(id)))
  }
}

// SET_EVENTS
export const setEvents = events => ({
  type: 'SET_EVENTS',
  events
})

export const startSetEvents = () => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid
    const events = []

    return db.collectionGroup('events').where("user", "==", uid).get().then(querySnapshot => {
      querySnapshot.forEach(snapshot => {
        events.push({
          id: snapshot.id,
          tripId: snapshot.ref.parent.parent.id,
          user: uid,
          ...snapshot.data()
        })
      })
      dispatch(setEvents(events))
    })
  }
}