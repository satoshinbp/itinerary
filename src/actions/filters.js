export const showUpcomingTrip = () => ({
  type: 'SHOW_UPCOMING_TRIP',
})

export const showPastTrip = () => ({
  type: 'SHOW_PAST_TRIP',
})

export const showAllTrip = () => ({
  type: 'SHOW_ALL_TRIP',
})