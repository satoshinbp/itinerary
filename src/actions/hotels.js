export const addHotel = (
  {
    id='',
    tripId = '',
    name = '',
    checkInDate = 0,
    checkOutDate = 0,
    checkInTime = 0,
    checkOutTime = 0,
    ETA = 0,
    ETD = 0,
    location ='',
    note = ''
  } = {}
) => ({
  type: 'ADD_HOTEL',
  hotel: {
    id,
    tripId,
    name,
    checkInDate,
    checkOutDate,
    checkInTime,
    checkOutTime,
    ETA,
    ETD,
    location,
    note
  }
})

export const editHotel = (id, updates) => ({
  type: 'EDIT_HOTEL',
  id,
  updates
})

export const removeHotel = (id = '') => ({
  type: 'REMOVE_HOTEL',
  id
})