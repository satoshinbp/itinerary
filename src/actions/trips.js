import db from '../firebase/firebase'

// ADD_TRIP
export const addTrip = trip => ({
  type: 'ADD_TRIP',
  trip
})

export const startAddTrip = (tripData = {}) => {
  return dispatch => {
    const {
      title = '',
      startDate = 0,
      endDate = 0,
      note = ''
    } = tripData
    const trip = { title, startDate, endDate, note }

    return db.collection('trips').add(trip).then(snapshot => {
      dispatch(addTrip({
        id: snapshot.id,
        ...trip
      }))
    })
  }
}

// EDIT_TRIP
export const editTrip = (id, updates) => ({
  type: 'EDIT_TRIP',
  id,
  updates
})

export const startEditTrip = (id, updates) => {
  return dispatch => {
    return db.collection('trips').doc(id).update(updates).then(() => {
      dispatch(editTrip(id, updates))
    })
  }
}

// REMOVE_TRIP
export const removeTrip = id => ({
  type: 'REMOVE_TRIP',
  id
})

export const startRemoveTrip = (id = '') => {
  return dispatch => {
    return db.collection('trips').doc(id).delete().then(() => {
      dispatch(removeTrip(id))
    })
  }
}

//SET_TRIPS
export const setTrips = trips => ({
  type: 'SET_TRIPS',
  trips
})

export const startSetTrips = () => {
  return dispatch => {
    return db.collection('trips').get().then(snapshot => {
      const trips = []

      snapshot.forEach(childSnapshot => {
        trips.push({
          id: childSnapshot.id,
          ...childSnapshot.data()
        })
      })

      dispatch(setTrips(trips))
    })
  }
}