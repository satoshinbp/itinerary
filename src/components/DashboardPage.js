import React from 'react';
import { Link } from 'react-router-dom';
import TripList from './TripList';
import TripListFilters from './TripListFilters';

const DashboardPage = () => (
  <div className="container">
    <div className="page-header">
      <Link className="btn-push" to="/create">Add Trip</Link>
      <TripListFilters />
    </div>
    <TripList />
  </div>
);

export default DashboardPage;