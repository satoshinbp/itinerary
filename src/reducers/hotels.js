const hotelsReducerDefaultState = []

export default (state = hotelsReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_HOTEL':
      return [
        ...state,
        action.hotel
      ]
    case 'EDIT_HOTEL':
      return state.map((hotel) => {
        if (hotel.id === action.id) {
          return {
            ...hotel,
            ...action.updates
          }
        } else {
          return hotel
        }
      })
    case 'REMOVE_HOTEL':
      return state.filter(({ id }) => id !== action.id)
    default:
      return state
  }
}