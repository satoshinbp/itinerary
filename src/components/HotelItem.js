import React from 'react'
import { connect } from 'react-redux'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import moment from 'moment'
import { removeHotel } from '../actions/hotels'

export const HotelItem = ({ dispatch, date, inOrOut, id, name, checkInDate, checkOutDate, checkInTime, checkOutTime, location, setHotelId }) => {
  const confirmRemoveHotel = () => {
    const result = confirm('Are you sure you want to delete this hotel?');

    if (result) {
      dispatch(removeHotel(id))
    }
  }

  return (
    <div className="schedule-list-item">
      <div className=" schedule-list-item__actions">
        <FontAwesomeIcon className="icon" icon="trash-alt" onClick={confirmRemoveHotel} />
      </div>
      <div className="schedule-list-item__time" onClick={() => setHotelId(id)}>
        {inOrOut === "in" && <FontAwesomeIcon className="icon show-for-desktop" icon="moon" />}
        {inOrOut === "out" && <FontAwesomeIcon className="icon show-for-desktop" icon="sun" />}
        {date.isSame(moment(checkInDate), 'day') && `Check In: ${moment(checkInTime).format('HH:mm')}~`}
        {date.isSame(moment(checkOutDate), 'day') && `Check Out: ~${moment(checkOutTime).format('HH:mm')}`}
      </div>
      <div className="schedule-list-item__title" onClick={() => setHotelId(id)}>
        <FontAwesomeIcon className="icon show-for-desktop" icon="hotel" />
        {name}
      </div>
      <div className="schedule-list-item__location" onClick={() => setHotelId(id)}>
        {location && <FontAwesomeIcon className="icon show-for-desktop" icon="map-marker-alt" />}
        {location}
      </div>
    </div>
  )
}

export default connect()(HotelItem);