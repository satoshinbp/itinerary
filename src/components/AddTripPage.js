import React from 'react';
import { connect } from 'react-redux';
import TripForm from './TripForm';
import { addTrip } from '../actions/trips';

const AddTripPage = (props) => (
  <div className="container">
    <TripForm
      onSubmit={(trip) => {
        props.dispatch(addTrip(trip));
        props.history.push('/');
      }} />
  </div>
);

export default connect()(AddTripPage);