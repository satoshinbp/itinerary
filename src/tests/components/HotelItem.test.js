import React from 'react'
import { shallow } from 'enzyme'
import moment from 'moment'
import { removeHotel } from '../../actions/hotels'
import { HotelItem } from '../../components/HotelItem'
import hotels from '../fixtures/hotels'

let dispatch, setHotelId, inOrOut, hotel, wrapper
const date = moment().add(1, 'day')

describe('for 2nd day of trips[0], check in hotels[0]', () => {
  beforeEach(() => {
    dispatch = jest.fn()
    setHotelId = jest.fn()
    inOrOut = "in"
    hotel = hotels[0]
    wrapper = shallow(<HotelItem dispatch={dispatch} date={date} {...hotel} setHotelId={setHotelId} inOrOut={inOrOut} />)
  })

  test('should render HotelItem correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should remove hotel when trash icon clicked', () => {
    window.confirm = jest.fn(() => true)
    wrapper.find({ icon: 'trash-alt' }).simulate('click')
    expect(dispatch).toBeCalledWith(removeHotel(hotel.id))
  })

  test('should set hotel id trip on click time', () => {
    wrapper.find(".schedule-list-item__time").simulate('click')
    expect(setHotelId).toBeCalledWith(hotel.id)
  })

  test('should set hotel id trip on click title', () => {
    wrapper.find(".schedule-list-item__title").simulate('click')
    expect(setHotelId).toBeCalledWith(hotel.id)
  })

  test('should set hotel id trip on click location', () => {
    wrapper.find(".schedule-list-item__location").simulate('click')
    expect(setHotelId).toBeCalledWith(hotel.id)
  })
})

describe('for 2nd day of trips[0], check out hotels[2]', () => {
  beforeEach(() => {
    dispatch = jest.fn()
    setHotelId = jest.fn()
    inOrOut = "out"
    hotel = hotels[2]
    wrapper = shallow(<HotelItem dispatch={dispatch} date={date} {...hotel} setHotelId={setHotelId} inOrOut={inOrOut} />)
  })

  test('should render HotelItem correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should remove hotel when trash icon clicked', () => {
    window.confirm = jest.fn(() => true)
    wrapper.find({ icon: 'trash-alt' }).simulate('click')
    expect(dispatch).toBeCalledWith(removeHotel(hotel.id))
  })

  test('should set hotel id trip on click time', () => {
    wrapper.find(".schedule-list-item__time").simulate('click')
    expect(setHotelId).toBeCalledWith(hotel.id)
  })

  test('should set hotel id trip on click title', () => {
    wrapper.find(".schedule-list-item__title").simulate('click')
    expect(setHotelId).toBeCalledWith(hotel.id)
  })

  test('should set hotel id trip on click location', () => {
    wrapper.find(".schedule-list-item__location").simulate('click')
    expect(setHotelId).toBeCalledWith(hotel.id)
  })
})

describe('for 2nd day of trips[0], stay in hotels[3], morning', () => {
  beforeEach(() => {
    dispatch = jest.fn()
    setHotelId = jest.fn()
    inOrOut = "in"
    hotel = hotels[3]
    wrapper = shallow(<HotelItem dispatch={dispatch} date={date} {...hotel} setHotelId={setHotelId} inOrOut={inOrOut} />)
  })

  test('should render HotelItem correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should remove hotel when trash icon clicked', () => {
    window.confirm = jest.fn(() => true)
    wrapper.find({ icon: 'trash-alt' }).simulate('click')
    expect(dispatch).toBeCalledWith(removeHotel(hotel.id))
  })

  test('should set hotel id trip on click time', () => {
    wrapper.find(".schedule-list-item__time").simulate('click')
    expect(setHotelId).toBeCalledWith(hotel.id)
  })

  test('should set hotel id trip on click title', () => {
    wrapper.find(".schedule-list-item__title").simulate('click')
    expect(setHotelId).toBeCalledWith(hotel.id)
  })

  test('should set hotel id trip on click location', () => {
    wrapper.find(".schedule-list-item__location").simulate('click')
    expect(setHotelId).toBeCalledWith(hotel.id)
  })
})

describe('for 2nd day of trips[0], stay in hotels[3], night', () => {
  beforeEach(() => {
    dispatch = jest.fn()
    setHotelId = jest.fn()
    inOrOut = "out"
    hotel = hotels[3]
    wrapper = shallow(<HotelItem dispatch={dispatch} date={date} {...hotel} setHotelId={setHotelId} inOrOut={inOrOut} />)
  })

  test('should render HotelItem correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should remove hotel when trash icon clicked', () => {
    window.confirm = jest.fn(() => true)
    wrapper.find({ icon: 'trash-alt' }).simulate('click')
    expect(dispatch).toBeCalledWith(removeHotel(hotel.id))
  })

  test('should set hotel id trip on click time', () => {
    wrapper.find(".schedule-list-item__time").simulate('click')
    expect(setHotelId).toBeCalledWith(hotel.id)
  })

  test('should set hotel id trip on click title', () => {
    wrapper.find(".schedule-list-item__title").simulate('click')
    expect(setHotelId).toBeCalledWith(hotel.id)
  })

  test('should set hotel id trip on click location', () => {
    wrapper.find(".schedule-list-item__location").simulate('click')
    expect(setHotelId).toBeCalledWith(hotel.id)
  })
})