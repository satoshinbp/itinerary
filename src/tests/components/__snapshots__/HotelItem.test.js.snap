// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`for 2nd day of trips[0], check in hotels[0] should render HotelItem correctly 1`] = `
<div
  className="schedule-list-item"
>
  <div
    className=" schedule-list-item__actions"
  >
    <FontAwesomeIcon
      border={false}
      className="icon"
      fixedWidth={false}
      flip={null}
      icon="trash-alt"
      inverse={false}
      listItem={false}
      mask={null}
      onClick={[Function]}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </div>
  <div
    className="schedule-list-item__time"
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className="icon show-for-desktop"
      fixedWidth={false}
      flip={null}
      icon="moon"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    Check In: 15:00~
  </div>
  <div
    className="schedule-list-item__title"
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className="icon show-for-desktop"
      fixedWidth={false}
      flip={null}
      icon="hotel"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    Waikiki Resort Hotel
  </div>
  <div
    className="schedule-list-item__location"
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className="icon show-for-desktop"
      fixedWidth={false}
      flip={null}
      icon="map-marker-alt"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    Waikiki
  </div>
</div>
`;

exports[`for 2nd day of trips[0], check out hotels[2] should render HotelItem correctly 1`] = `
<div
  className="schedule-list-item"
>
  <div
    className=" schedule-list-item__actions"
  >
    <FontAwesomeIcon
      border={false}
      className="icon"
      fixedWidth={false}
      flip={null}
      icon="trash-alt"
      inverse={false}
      listItem={false}
      mask={null}
      onClick={[Function]}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </div>
  <div
    className="schedule-list-item__time"
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className="icon show-for-desktop"
      fixedWidth={false}
      flip={null}
      icon="sun"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    Check Out: ~10:00
  </div>
  <div
    className="schedule-list-item__title"
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className="icon show-for-desktop"
      fixedWidth={false}
      flip={null}
      icon="hotel"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    Hawai Travellars Inn
  </div>
  <div
    className="schedule-list-item__location"
    onClick={[Function]}
  />
</div>
`;

exports[`for 2nd day of trips[0], stay in hotels[3], morning should render HotelItem correctly 1`] = `
<div
  className="schedule-list-item"
>
  <div
    className=" schedule-list-item__actions"
  >
    <FontAwesomeIcon
      border={false}
      className="icon"
      fixedWidth={false}
      flip={null}
      icon="trash-alt"
      inverse={false}
      listItem={false}
      mask={null}
      onClick={[Function]}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </div>
  <div
    className="schedule-list-item__time"
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className="icon show-for-desktop"
      fixedWidth={false}
      flip={null}
      icon="moon"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </div>
  <div
    className="schedule-list-item__title"
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className="icon show-for-desktop"
      fixedWidth={false}
      flip={null}
      icon="hotel"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    Hawai Hotel
  </div>
  <div
    className="schedule-list-item__location"
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className="icon show-for-desktop"
      fixedWidth={false}
      flip={null}
      icon="map-marker-alt"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    Hawai
  </div>
</div>
`;

exports[`for 2nd day of trips[0], stay in hotels[3], night should render HotelItem correctly 1`] = `
<div
  className="schedule-list-item"
>
  <div
    className=" schedule-list-item__actions"
  >
    <FontAwesomeIcon
      border={false}
      className="icon"
      fixedWidth={false}
      flip={null}
      icon="trash-alt"
      inverse={false}
      listItem={false}
      mask={null}
      onClick={[Function]}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </div>
  <div
    className="schedule-list-item__time"
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className="icon show-for-desktop"
      fixedWidth={false}
      flip={null}
      icon="sun"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </div>
  <div
    className="schedule-list-item__title"
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className="icon show-for-desktop"
      fixedWidth={false}
      flip={null}
      icon="hotel"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    Hawai Hotel
  </div>
  <div
    className="schedule-list-item__location"
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className="icon show-for-desktop"
      fixedWidth={false}
      flip={null}
      icon="map-marker-alt"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    Hawai
  </div>
</div>
`;
